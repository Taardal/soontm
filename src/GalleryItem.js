import React from "react";
import PropTypes from "prop-types";
import { TouchableHighlight, Image } from "react-native";
import { styles } from "./GalleryItem.styles";

const GalleryItem = ({ imageUrl, width, height, onClick }) => (
  <TouchableHighlight onPress={onClick}>
    <Image
      source={{ uri: imageUrl }}
      style={[
        styles.image,
        {
          width: width,
          height: height
        }
      ]}
    />
  </TouchableHighlight>
);

GalleryItem.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired
};

export default GalleryItem;
