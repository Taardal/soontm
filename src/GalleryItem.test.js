import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Enzyme from "enzyme";
import GalleryItem from "./GalleryItem";

describe("GalleryItem", () => {
  it("matches snapshot", () => {
    const shallowRenderer = new ShallowRenderer();

    shallowRenderer.render(
      <GalleryItem imageUrl={"imageUrl"} width={100} height={100} onClick={jest.fn()} />
    );
    const renderOutput = shallowRenderer.getRenderOutput();

    expect(renderOutput).toMatchSnapshot();
  });

  it("renders image with expected props", () => {
    const imageUrl = "imageUrl";
    const width = 100;
    const height = 100;

    const enzyme = Enzyme.shallow(
      <GalleryItem imageUrl={imageUrl} width={width} height={height} onClick={jest.fn()} />
    );

    const image = enzyme.find("Image");
    expect(image).toBeTruthy();
    expect(image.props().source.uri).toBe(imageUrl);

    const stylesUsingStyleSheet = image.props().style[0];
    const stylesUsingProps = image.props().style[1];
    expect(stylesUsingProps.width).toBe(width);
    expect(stylesUsingProps.height).toBe(height);
  });

  it("calls callback on press", () => {
    const onClick = jest.fn();
    
    const enzyme = Enzyme.shallow(<GalleryItem imageUrl={"imageUrl"} width={100} height={100} onClick={onClick} />);
    
    enzyme.find("TouchableHighlight").simulate("press");
    expect(onClick.mock.calls.length).toBe(1);
  });
});
