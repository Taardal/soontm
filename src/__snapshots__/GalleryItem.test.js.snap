// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem matches snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  onPress={[MockFunction]}
  underlayColor="black"
>
  <Image
    source={
      Object {
        "uri": "imageUrl",
      }
    }
    style={
      Array [
        Object {
          "margin": 0.5,
        },
        Object {
          "height": 100,
          "width": 100,
        },
      ]
    }
  />
</TouchableHighlight>
`;
